src/webui/www/private/views/preferences.html

```diff
@@ -953,6 +953,14 @@
                     <input type="checkbox" id="reannounceWhenAddressChanged" />
                 </td>
             </tr>
+            <tr>
+                <td>
+                    <label for="webUIRefreshInterval">QBT_TR(Web UI refresh interval:)QBT_TR[CONTEXT=OptionsDialog]</label>
```

Comment:
This should placed under WebUI tab (in both GUI and WebUI).

But before that, I think the current implementation is going the wrong way.
`serverState[KEY_SYNC_MAINDATA_REFRESH_INTERVAL] = session->refreshInterval();` is correct because that is what `"refresh_interval": torrents table refresh interval` represents, i.e. how regularly the session data/torrent table is updated. The webui would be sensible to have a refresh interval >= session refresh interval.
Now if you want to change the webui update interval then it should be done purely at webui side.

Reply:
> if you want to change the webui update interval then it should be done purely at webui side.

👍 
As I suggested above.

Reply:
I just want to change how often the webUI updates on the user's web browser. This may be independent from how often the session data  is updated. But we can also restrict the lower limit so that user can't set it lower than how fast the session is being updated.

>Now if you want to change the webui update interval then it should be done purely at webui side.
>

Do you mean I should implement the preference inside webUI code?

Reply:
>I just want to change how often the webUI updates on the user's web browser. 

Exactly, I understand.

>Do you mean I should implement the preference inside webUI code?

Yeah probably, only html + js and no c++ code changes.

Reply:
But I want to look at server side cost of updating/syncing the data too fast as well. The web UI doesn’t perform well when there’s a lot of torrents and the server keeps refreshing torrent table at such low intervals.

Reply:
There is already an option for that if I'm not mistaken.

Reply:
You mean by changing transfer list refresh interval from the GUI? But what if I want the GUI to update fast and not the Web UI.

Reply:
>But what if I want the GUI to update fast and not the Web UI.

Then you adjust the setting for webui.

Reply:
> > But what if I want the GUI to update fast and not the Web UI.
> 
> Then you adjust the webui setting to be slower and GUI setting to be faster.
>

There's no WEBUI settings to do that independently in qBittorrent currently(correct me if I'm wrong).

Also seems like this transfer list refresh interval is a wrong terminology used by the devs since actually you not only update the transfer list but the filters, tabs(general, speed,contents, peers) and loads of other things as well. Maybe it's time to rename this term and allow setting the refresh interval for other things independently because it gives the user more control over how fast the want to see certain things update. If you're aginst it then at least rename it atleast.

Reply:
> There's no WEBUI settings to do that independently in qBittorrent currently(correct me if I'm wrong).

Yes. Since it sticks to core update interval currently. If you want to update it independently then you need independent setting for it.

Reply:
@summerqB 
Actually it's the BitTorrent session refresh interval when it updates some cached data, analyzes it and signals about some events. Some UI elements are simply updated based on these signals.